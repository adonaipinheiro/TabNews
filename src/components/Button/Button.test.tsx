import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';

import {Button} from './';

jest.mock('@react-navigation/elements', () => ({
  getDefaultHeaderHeight: jest.fn(),
}));

describe('<Button />', () => {
  it('should renders button with text="Entrar"', () => {
    const {getByText} = render(<Button text="Entrar" />);
    const buttonText = getByText('Entrar');
    expect(buttonText).toBeTruthy();
  });

  it('should renders loading state correctly', () => {
    const {getByText} = render(
      <Button text="Entrar" loading={true} />,
    );
    const loadingText = getByText('Carregando');
    expect(loadingText).toBeTruthy();
  });

  it('should calls onPress callback when button is pressed', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button text="Entrar" onPress={onPressMock} />,
    );
    const button = getByText('Entrar');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalled();
  });

  it('should verify if button is disabled when loading is true', () => {
    const {getByText, root} = render(
      <Button text="Entrar" loading={true} />,
    );
    const button = getByText('Carregando');
    expect(button).toBeTruthy();
    expect(root.props.accessibilityState.disabled).toEqual(true);
  });

  it('should take a snapshot', () => {
    const {toJSON} = render(<Button text="Click me" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
