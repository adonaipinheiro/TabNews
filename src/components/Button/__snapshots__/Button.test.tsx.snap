// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should take a snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(31, 136, 61)",
      "borderRadius": 6,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 14,
        "fontWeight": "600",
      }
    }
  >
    Click me
  </Text>
</View>
`;
